aptos/token_bridge/sources/bridge_implementation.move

```diff
@@ -0,0 +1,138 @@
+    use token_bridge::bridge_structs::{Self, create_asset_meta, encode_asset_meta, AssetMeta};
+    use token_bridge::utils::{hash_type_info};
+
+    use wormhole::u256::{Self, U256};
+    //use wormhole::u128::{U128};
+    use wormhole::u32::{U32};
+    use wormhole::u16::{U16};
+    use wormhole::vaa::{Self, VAA, parse_and_verify};
+
+    use token_bridge::deploy_coin::{deploy_coin};
+
+
+    // TODO: for functions that do take a signer, we should have an equivalent
+    // function that does *not* take a signer, and instead takes explicitly
+    // whatever's needed from the signer. In this case, it would be a version of
+    // `attest_token` that takes a sufficient amount of coins. Then the signer
+    // version could do the withdrawal first and call the other version (which
+    // itself should also be public).
+    // There are multiple benefits:
+    // 1. we have a version of the function that doesn't require a signer
+    // 2. structuring the code in this way makes it very clear what the signer
+    //    is needed for (since the signer stuff is forced to be written separately)
+    //    -- something not that clear in the current version
+    public fun attest_token<CoinType>(user: &signer) {
+        let payload_id = 0;
+        let token_address = hash_type_info<CoinType>();
+        if (!bridge_state::is_registered_native_asset(token_address) && !bridge_state::is_wrapped_asset(token_address)) {
+            // if native asset is not registered, register it in the reverse look-up map
+            set_native_asset(token_address, type_of<CoinType>());
+        };
+        let token_chain = wormhole::state::get_chain_id();
+        let decimals = decimals<CoinType>();
+        let symbol = *string::bytes(&symbol<CoinType>());
+        let name = *string::bytes(&name<CoinType>());
+
+        let _asset_meta: AssetMeta = create_asset_meta(
+            payload_id,
+            token_address,
+            token_chain,
+            decimals,
+            symbol,
+            name
+        );
+
+        let payload:vector<u8> = encode_asset_meta(_asset_meta);
+        let nonce = 0;
+        let message_fee = wormhole::state::get_message_fee();
+        let fee_coins = withdraw(user, message_fee);
+        bridge_state::publish_message(
+            nonce,
+            payload,
+            fee_coins
+        )
+    }
+
+    public entry fun create_wrapped<CoinType>(vaa: vector<u8>): address {
+        let vaa = parse_and_verify(vaa);
+        let _token_bridge_signer = token_bridge_signer();
+        let seed = b"blah"; //TODO - how to set seed?
+        let (new_signer, new_cap) = create_resource_account(&_token_bridge_signer, seed);
+        let token_address = address_of(&new_signer);
+        deploy_coin(&new_signer);
+        bridge_state::set_wrapped_asset_signer_capability(to_bytes(&token_address), new_cap);
+
+        let _asset_meta: AssetMeta = bridge_structs::parse_asset_meta(vaa::get_payload(&vaa));
```

Comment:
why `vaa::get_payload` instead of `vaa:destroy` here? We don't need the VAA later anyway.